internal/db/db.go

```diff
@@ -143,9 +146,63 @@ func Connect(connStr string, overrides ...Overrideable) (*sqlx.DB, error) {
 	if err != nil {
 		return nil, fmt.Errorf("could not connect: %v", err)
 	}
-	db, err := sqlx.Connect("postgres", mcc.DSN())
+	db, err := sqlx.Connect("pgx", mcc.DSN())
 	if err != nil {
 		return nil, fmt.Errorf("could not connect: %v", err)
 	}
 	return db, nil
 }
+
+// CopyFromLines bulk-loads data using the given copyCmd. Returns the number of
+// rows inserted.
+func CopyFromLines(db *sqlx.DB, lines []string, copyCmd string) (int64, error) {
+	conn, err := db.Conn(context.Background())
+	if err != nil {
+		return 0, fmt.Errorf("acquiring DB connection for COPY: %w", err)
+	}
+	defer conn.Close()
+
+	// Our slice of lines needs to be presented to PgConn.CopyFrom() as a
+	// unified io.Reader. Joining the entire string would eat memory, so use an
+	// intermediary io.Pipe instead.
+	//
+	// TODO: just take the bytes directly as an argument rather than converting
+	// to a slice and back
+	r, w := io.Pipe()
+	go func() {
```

Comment:
This pipe approach seems to be causing a significant performance hit compared to `lib/pq`, so I'll look into fixing this TODO.